tools/azure-pipelines/jobs-template.yml

```diff
@@ -160,6 +176,13 @@ jobs:
       targetPath: $(DEBUG_FILES_OUTPUT_DIR)
       artifact: logs-${{parameters.stage_name}}-$(DEBUG_FILES_NAME)
 
+  - script: |
+      mkdir -p $(TESTCONTAINER_CACHE_FOLDER)
+      docker save $(docker image ls --format "{{.Repository}}:{{.Tag}}") -o $(TESTCONTAINER_CACHE_FOLDER)/cache.tar || true
```

Comment:
I'm wondering if we'd should filter some of these out. Even the [core stage](https://dev.azure.com/Autophagy/Flink-CI/_build/results?buildId=80&view=logs&j=9dc1b5dc-bcfa-5f83-eaa7-0cb181ddc267&t=95a75a7f-839b-5434-cae3-c0698e9852d8), which doesn't use test containers at all still caches 4-5gigs. (consuming 8 minutes in the process)

Reply:
Yeah, I think the generalised cache is also catching a lot of base images that just come with the build environment. We could limit this to testcontainers/*, elasticsearch, kafka, rabbitmq, mysql and postgres.